codegen/src/types.rs

```diff
@@ -1046,31 +1052,35 @@ pub struct FileDescriptor {
 
 impl FileDescriptor {
 
-    pub fn write_proto<P: AsRef<Path>>(in_file: P, out_file: P) -> Result<()> {
-        let mut desc = FileDescriptor::read_proto(&in_file)?;
-        desc.fetch_imports(in_file.as_ref())?;
+    pub fn write_proto<P: AsRef<Path>>(config: &Config<P>) -> Result<()> {
```

Comment:
I think this fn could go in `Config` instead. 
Then we could probably have `FileDescriptor` private.